editor/inserter/menu.js

```diff
@@ -58,37 +56,66 @@ class InserterMenu extends Component {
 		} );
 	}
 
-	selectBlock( name ) {
+	selectBlock( blockType ) {
 		return () => {
-			this.props.onSelect( name );
+			this.props.onSelect( blockType.name );
+			recordUsage( blockType );
```

Comment:
Yes, that's a much better way.

As that is the beginning of moving things into the actions and app state instead of a module level variable, I'll mark this PR as WIP until that's done and using the REST API to keep a record of block usage.